internal/oci/state.go

```diff
@@ -6,15 +6,25 @@ import (
 	"os"
 	"path/filepath"
 
-	"github.com/arm/remoteproc-runtime/internal/rootpath"
+	"github.com/arm/remoteproc-runtime/internal/userdirs"
 	"github.com/opencontainers/runtime-spec/specs-go"
 )
 
 const (
 	stateFileName = "state.json"
 )
 
-var stateDir = rootpath.Join("run", "remoteproc")
+var (
+	stateDir    string
+	stateDirErr error
+)
+
+func init() {
+	stateDir, stateDirErr = userdirs.RuntimeDir()
+	if stateDirErr != nil || stateDir == "" {
```

Comment:
If `err != nil`, we should be able to safely assume that `stateDir` is a legit value ready to use. In other words, we shouldn't test if it's empty here in addition to testing the error.